lms/static/js/edxnotes/views/search_box.js

```diff
@@ -1,22 +1,22 @@
 ;(function (define, undefined) {
 'use strict';
 define([
-    'jquery', 'underscore', 'backbone', 'gettext', 'js/edxnotes/utils/logger',
+    'jquery', 'underscore', 'backbone', 'gettext', 'edx-ui-toolkit/js/utils/html-utils', 'js/edxnotes/utils/logger',
     'js/edxnotes/collections/notes'
-], function ($, _, Backbone, gettext, NotesLogger, NotesCollection) {
+], function ($, _, Backbone, gettext, HtmlUtils, NotesLogger, NotesCollection) {
     var SearchBoxView = Backbone.View.extend({
         events: {
             'submit': 'submitHandler'
         },
 
         errorMessage: gettext('An error has occurred. Make sure that you are connected to the Internet, and then try refreshing the page.'),
         emptyFieldMessage: (function () {
-            var message = gettext('Please enter a term in the %(anchor_start)s search field%(anchor_end)s.');
-            return interpolate(message, {
-                'anchor_start': '<a href="#search-notes-input">',
-                'anchor_end': '</a>'
-            }, true);
-        } ()),
+            var message = gettext('Please enter a term in the {anchorStart} search field{anchorEnd}.');
+            return HtmlUtils.interpolateHtml(message, {
+                anchorStart: HtmlUtils.HTML('<a href="#search-notes-input">'),
+                anchorEnd: HtmlUtils.HTML('</a>')
+            }).text;
```

Comment:
The problem is that once we pass around escaped HTML that isn't wrapped as an HTML snippet, then the code that eventually renders it is unsafe. In this case, the message is passed to an error handler:

https://github.com/edx/edx-platform/blob/master/lms/static/js/edxnotes/views/tabs/search_results.js#L144

which in turn passes the string to this function which unsafely renders it as HTML:

https://github.com/edx/edx-platform/blob/master/lms/static/js/edxnotes/views/tab_view.js#L124

We need to fix this last function to use a safe function, which will then end up double-escaping the HTML. The intended approach is that any correctly escaped HTML is wrapped as an HTML snippet, and it is only rendered to a page using an HtmlUtils function that knows how to handle it.